src/legacy/server/kbn_server.d.ts

```diff
@@ -101,38 +105,63 @@ export interface PluginsSetup {
   [key: string]: object;
 }
 
+export interface Platform {
+  env: {
+    mode: Readonly<EnvironmentMode>;
+    packageInfo: Readonly<PackageInfo>;
+  };
+  handledConfigPaths: UnwrapPromise<ReturnType<ConfigService['getUsedPaths']>>;
+  setupDeps: {
+    core: CoreSetup;
+    plugins: PluginsSetup;
+  };
+  startDeps: {
+    core: CoreSetup;
+    plugins: Record<string, object>;
+  };
+  __internals: {
+    elasticsearch: LegacyServiceSetupDeps['core']['elasticsearch'];
+    hapiServer: LegacyServiceSetupDeps['core']['http']['server'];
+    kibanaMigrator: LegacyServiceStartDeps['core']['savedObjects']['migrator'];
+    savedObjectsClientProvider: LegacyServiceStartDeps['core']['savedObjects']['clientProvider'];
+    uiPlugins: LegacyServiceSetupDeps['core']['plugins']['uiPlugins'];
+    uiSettings: LegacyServiceSetupDeps['core']['uiSettings'];
+    rendering: LegacyServiceSetupDeps['core']['rendering'];
+  };
+  logger: LoggerFactory;
+}
+
+export interface LegacyPlugins {
+  pluginSpecs: LegacyServiceDiscoverPlugins['pluginSpecs'];
+  disabledPluginSpecs: LegacyServiceDiscoverPlugins['pluginSpecs'];
+  uiExports: LegacyServiceDiscoverPlugins['uiExports'];
+}
+
 // eslint-disable-next-line import/no-default-export
 export default class KbnServer {
   public readonly newPlatform: {
-    __internals: {
-      hapiServer: LegacyServiceSetupDeps['core']['http']['server'];
-      uiPlugins: LegacyServiceSetupDeps['core']['plugins']['uiPlugins'];
-      elasticsearch: LegacyServiceSetupDeps['core']['elasticsearch'];
-      uiSettings: LegacyServiceSetupDeps['core']['uiSettings'];
-      kibanaMigrator: LegacyServiceStartDeps['core']['savedObjects']['migrator'];
-    };
-    env: {
-      mode: Readonly<EnvironmentMode>;
-      packageInfo: Readonly<PackageInfo>;
-    };
+    __internals: Omit<Platform['__internals'], 'savedObjectsClientProvider'>;
+    env: Platform['env'];
     coreContext: {
-      logger: LoggerFactory;
-    };
-    setup: {
-      core: CoreSetup;
-      plugins: PluginsSetup;
-    };
-    start: {
-      core: CoreStart;
-      plugins: Record<string, object>;
+      logger: Platform['logger'];
     };
+    setup: Platform['setupDeps'];
+    start: Platform['startDeps'];
     stop: null;
+    params: {
+      handledConfigPaths: Platform['handledConfigPaths'];
```

Comment:
wasn't it deleted in master?

Reply:
It was, in https://github.com/elastic/kibana/pull/51880